onboarding/main.go

```diff
@@ -155,21 +138,11 @@ func convertRsaPublicKeyAsPemStr(publicKey *rsa.PublicKey) string {
 	return string(publicKeyPem)
 }
 
-func getStorageClusterMetadata(ctx context.Context, operatorNamespace string, clientset *kubernetes.Clientset) (*metav1.PartialObjectMetadata, error) {
-	var storageClusterMetadata metav1.PartialObjectMetadata
-	storageClusterGVKPath := fmt.Sprintf(
-		"/apis/ocs.openshift.io/v1/namespaces/%s/storageclusters/%s",
-		operatorNamespace,
-		storageClusterName,
-	)
-	storageClusterMetadataJSON, err := clientset.RESTClient().Get().AbsPath(storageClusterGVKPath).Do(ctx).Raw()
-	if err != nil {
-		return nil, fmt.Errorf("failed to get storage cluster metadata: %v", err)
-	}
-
-	if err = json.Unmarshal(storageClusterMetadataJSON, &storageClusterMetadata); err != nil {
-		return nil, fmt.Errorf("failed to parse storage cluster metadata response: %v", err)
+func getStorageCluster(ctx context.Context, operatorNamespace string, client client.Client) (*v1.StorageCluster, error) {
+	storageCluster := &v1.StorageCluster{}
+	if err := client.Get(ctx, types.NamespacedName{Name: storageClusterName, Namespace: operatorNamespace}, storageCluster); err != nil {
+		return nil, err
 	}
 
-	return &storageClusterMetadata, nil
+	return storageCluster, nil
```

Comment:
Why do we need this func if the function only executes a single Get command? Let's move the implementation to the call site.